apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadJsonOptions.cs

```diff
@@ -77,8 +77,16 @@ public sealed class UploadJsonOptions : JobCreationOptions
         public SchemaUpdateOption? DestinationSchemaUpdateOptions { get; set; }
 
         /// <summary>
-        /// Delegate that is used to modify the underlying <see cref="JobConfigurationLoad"/>.
+        /// Optional action to perform after preparing the request. If this property is non-null,
+        /// the <see cref="JobConfigurationLoad"/> used for a request will be passed to the delegate
+        /// before the request is executed. This allows for fine-grained modifications which aren't
+        /// otherwise directly supported by the properties in this options type.
         /// </summary>
+        /// <remarks>
+        /// Properties within this type are used to prepare the request before the delegate is called,
```

Comment:
It looks good. 👍🏼 This conveys the message and helps me use this property correctly. 
I think we can break the sentence for even better clarity. Something like:
```Properties within this type are used to prepare the request before the delegate is called. Therefore, the delegate should generally not modify .........```
But it is fine without that change as well.

Reply:
I think something along the lines of preference would be clearer:

"Prefer the properties on this type over this modifier to prepare the request. Only use this modifier to configure aspects for which there are no properties available. This modifier is applied to the request after all properties on this type have been applied."

Reply:
@amanda-tarafa: Works for me. So that's just changing the first sentence in the remarks, and I'm okay to keep the one about "The delegate is only called once [...]"?

Reply:
Yes, exactly. And also yes, keep the one about the delegate being called only once.